### ISSUE TITLE
end rewrite of the wrong column after complicated chain of corrections

### ISSUE THREAD

**Issue by Dorian** [rubocop-hq/rubocop#3980]
## Expected behavior

The `end` is placed correctly:

```ruby
def f
  a =
    if b
      c
    else
      d.each do |e|
        if e
          f
        else
          g
        end
      end
    end

  puts a
end
```

## Actual behavior

The `end` if misplaced:

```ruby
def f
  a =
    if b
      c
    else
      d.each do |e|
        if e
          f
        else
          g
        end
      end
         end

  puts a
end
```

## Steps to reproduce the problem

Have a file named `script.rb`:

```ruby
def f
  if b
    a = c
  else
    a = \
      d.each do |e|
        if e
          f
        else
          g
        end
      end
  end

  puts a
end
```

Then run `bundle exec rubocop -a script.rb`

## RuboCop version

```
$ rubocop -V
0.47.1 (using Parser 2.3.3.1, running on ruby 2.3.3 x86_64-darwin16)
```

**Comment by backus** [comment]
If you are talking about AutoCorrect behavior for EndAlignment cops then this is a dupe of https://github.com/bbatsov/rubocop/issues/3900, https://github.com/bbatsov/rubocop/issues/3866, and some other issues probably.

**Comment by koic** [comment]
It does not reproduce with the latest RuboCop 0.57.2. The position of `end` is auto-corrected with the same indent as `if` and `else`. Please give feedback if there is a problem again. Thanks.
